SPO/README.md

```diff
@@ -0,0 +1,116 @@
+
+## Create an Application Insights Resource and find your instrumentationKey
+Please find the instructions [here](https://docs.microsoft.com/en-us/azure/azure-monitor/app/create-new-resource).
+
+## Set up Application Insights 
+1. Follow [SPFx Extensions instruction](https://docs.microsoft.com/en-us/sharepoint/dev/spfx/extensions/get-started/build-a-hello-world-extension) to create a SharePoint Framework Extension.
+2. Install Application Insights via [NPM setup](https://github.com/microsoft/ApplicationInsights-JS#getting-started). 
+3. Under your Extension solution src folder, update `ApplicationCustomizer.ts` file:
+
+    a. [Set up Application Insights](https://github.com/microsoft/ApplicationInsights-JS#npm-setup-ignore-if-using-snippet-setup)
+    ```js
+    import { ApplicationInsights } from '@microsoft/applicationinsights-web'
+
+    const appInsights = new ApplicationInsights({ config: {
+    instrumentationKey: 'YOUR_INSTRUMENTATION_KEY_GOES_HERE'
+    /* ...Other Configuration Options... */
+    } });
+    appInsights.loadAppInsights();
+    appInsights.trackPageView(); // Manually call trackPageView to establish the current user/session/pageview
+    ```
+
+    b. Expose all configuration options you need in the ApplicationCustomizerProperties interface.
+    ```js
+    export interface IAppInsightsApplicationCustomizerProperties {
+        /* ...Other Configuration Options... */
+        instrumentationKey: string;
+    }
+    ```
+    c. Update appInsights instance config to map with exposed properties.
+    ```js
+    import { ApplicationInsights } from '@microsoft/applicationinsights-web'
+
+    const appInsights = new ApplicationInsights({ config: {
+    instrumentationKey: this.properties.instrumentationKey
+    /* ...Other Configuration Options... */
+    } });
+    appInsights.loadAppInsights();
+    appInsights.trackPageView(); // Manually call trackPageView to establish the current user/session/pageview
+    ```
+## Set up deployment configuration
+### Select Option A or Option B for deployment based on your needs: 
+The custom properties (eg. instrumentationKey in this example) is stored in several locations: 
+1. When running a gulp serve and running the page in debug mode, the key that is stored in `config\serve.json` -> `properties` is passed to the page as querystring.
+2. When the extension is installed on a site, the key in `sharepoint\assets\elements.xml` -> `ClientSideComponentProperties` is used (Option B).
+3. When deployed tenant wide, the key in `sharepoint\assets\ClientSideInstance.xml` -> `Properties` is used (Option A).
+
+The [AppInsightsExtensionSolutionSample](https://github.com/microsoft/ApplicationInsights-JS/tree/master/SPO/AppInsightsExtensionSolutionSample) is using Option A. We highly recommend you to create your own SPFx Extension solution, in order to get the full support of Application Insights and configure the SDK based on your needs.
```

Comment:
@DawgFan hi Matt, do you think this section works?

Reply:
Yes, definitely helps! Thank you!